### PR TITLE
Remove compiler args

### PR DESCRIPTION
**Pull request by sabel83**

### CODE REVIEW

File: lib/core/parse_config.cpp [hunk 1]
```diff
@@ -338,3 +371,21 @@ bool parse_config_result::should_error_at_exit() const
...
+  switch (a_)
+  {
+  case parse_config_result::action_t::run_shell:
+    return out_ << "run_shell";
+  case parse_config_result::action_t::exit_with_error:
+    return out_ << "exit_with_error";
+  case parse_config_result::action_t::exit_without_error:
+    return out_ << "exit_without_error";
+  default:
+    assert(!"Invalid action");
```
Comment (r0mai) [inline lib/core/parse_config.cpp]: return out_ here in case asserts are turned off.

**Comment by r0mai** [comment]
See comment, otherwise looks nice :+1:

File: lib/core/parse_config.cpp (hunk 1 above)
Comment (sabel83) [inline lib/core/parse_config.cpp]: I've added std::abort() there. assert(false) means that if the execution gets there, the program is in an invalid state ("this should never happen" :))

Comment (r0mai) [inline lib/core/parse_config.cpp]: We should have an `unreachable` macro :) but it's fine like this for now

**Pull request merged**